contracts/RiskFund/RiskFund.sol

```diff
@@ -260,48 +261,47 @@ contract RiskFund is AccessControlledV8, ExponentialNoError, ReserveHelpers, Max
         address[] calldata path
     ) internal returns (uint256) {
         require(amountOutMin != 0, "RiskFund: amountOutMin must be greater than 0 to swap vToken");
-        require(amountOutMin >= minAmountToConvert, "RiskFund: amountOutMin should be greater than minAmountToConvert");
         uint256 totalAmount;
 
         address underlyingAsset = vToken.underlying();
         address convertibleBaseAsset_ = convertibleBaseAsset;
         uint256 balanceOfUnderlyingAsset = _poolsAssetsReserves[comptroller][underlyingAsset];
 
-        ComptrollerViewInterface(comptroller).oracle().updatePrice(address(vToken));
+        if (balanceOfUnderlyingAsset == 0) {
+            return 0;
+        }
 
-        uint256 underlyingAssetPrice = ComptrollerViewInterface(comptroller).oracle().getUnderlyingPrice(
-            address(vToken)
-        );
+        ResilientOracleInterface oracle = ComptrollerViewInterface(comptroller).oracle();
+        oracle.updateAssetPrice(convertibleBaseAsset_);
+        Exp memory  baseAssetPrice = Exp({ mantissa: oracle.getPrice(convertibleBaseAsset_) });
+        uint256 amountOutMinInUsd = mul_ScalarTruncate(baseAssetPrice, amountOutMin);
+
+        if (amountOutMinInUsd < minAmountToConvert) {
+            return 0;
```

Comment:
I think this was the idea, yeah. I didn't change the behavior in this regard. But I think reverting should be fine as well, maybe it would better match the expectations of the callers (e.g. imagine the oracle price has changed – we'd probably want to fail rather than "do other swaps and let the user check which ones were indeed executed"). LMK what you think.

Reply:
It's ok to return 0 if `balanceOfUnderlyingAsset == 0`, but I would stop the TX in the `minAmountToConvert` check if it's unsatisfied.

Initially, we this require:
```
require(amountOutMin >= minAmountToConvert, "RiskFund: amountOutMin should be greater than minAmountToConvert");
```

I know the units were wrong, but it was stopping the TX

Reply:
https://github.com/VenusProtocol/isolated-pools/pull/287/commits/9930fd5298ddc677992b28360fc6eb7931ff6dcd